doc/introduction.rst

```diff
@@ -79,6 +79,34 @@ necessarily the identity of indiscernibles.
   parameterizations are equivalent. In practice, an algorithm may thus solve
   the metric learning problem with respect to either :math:`M` or :math:`L`.
 
+.. _bilinear_similarity:
+
+Bilinear Similarity
+===================
+
+Some algorithms in the package don't learn a distance or pseudo-distance, but
+a similarity. The idea is that two pairs are closer if their similarity value
+is high, and viceversa. Given a real-valued parameter matrix :math:`W` of shape
+``(n_features, n_features)`` where ``n_features`` is the number features
+describing the data, the Bilinear Similarity associated with :math:`W` is
+defined as follows:
+
+.. math:: S_W(x, x') = x^T W x'
+
+The matrix :math:`W` is not required to be positive semi-definite (PSD), so
+none of the distance properties are satisfied: nonnegativity, identity of
+indiscernibles, symmetry and triangle inequality.
+
+This allows some algorithms to optimize :math:`S_W` in an online manner using a
+simple and efficient procedure, and thus can be applied to problems with
+millions of training instances and achieves state-of-the-art performance
+on an image search task using :math:`k`-NN.
```

Comment:
this is specific to OASIS. nothing says that these things are not possible also for Mahalanobis distances

I would remove this entirely and replace by a sentence at the beginning of the next paragraph: "The absence of PSD constraint can enable the design of more efficient algorithms."
(and then "It also...")

Reply:
Yeah, I took that paragraph from the Metric Learning book, I guess it was the paragraph in between Bilinear Similarity and OASIS so that's why it's so specific. I removed it and added the sentence you mention in the next paragraph.